File: src/parent.js
const net = require('net');
const vsock = require('vsock');

// 连接到 enclave
const client = vsock.connect(8000, () => {
    console.log('已连接到 enclave');

    // 准备加密数据（这里使用示例数据，实际使用时需要替换为真实的加密数据）
    const encryptedData = Buffer.from('示例加密数据').toString('base64');

    // 构建请求
    const request = {
        encrypted_data: encryptedData
    };

    // 发送请求
    client.write(JSON.stringify(request));
    client.end();
});

// 接收响应
let data = '';
client.on('data', (chunk) => {
    data += chunk;
});

client.on('end', () => {
    try {
        const response = JSON.parse(data);
        console.log('解密后的数据:', response.decrypted_data);
        console.log('公钥:', response.public_key);
    } catch (error) {
        console.error('处理响应时出错:', error);
    }
}); 

File: src/parent.rs
use anyhow::Result;
use serde::{Deserialize, Serialize};
use std::io::{Read, Write};
use vsock::{VsockAddr, VsockStream};
use base64::{engine::general_purpose::STANDARD as BASE64, Engine};

#[derive(Debug, Serialize, Deserialize)]
struct DecryptRequest {
    encrypted_data: String,
}

#[derive(Debug, Serialize, Deserialize)]
struct DecryptResponse {
    decrypted_data: String,
    public_key: String,
}

fn main() -> Result<()> {
    // 连接到 enclave
    let mut stream = VsockStream::connect(&VsockAddr::new(8000))?;
    println!("已连接到 enclave");

    // 准备加密数据（这里使用示例数据，实际使用时需要替换为真实的加密数据）
    let encrypted_data = "示例加密数据";
    let encrypted_base64 = BASE64.encode(encrypted_data);

    // 构建请求
    let request = DecryptRequest {
        encrypted_data: encrypted_base64,
    };

    // 发送请求
    let request_json = serde_json::to_vec(&request)?;
    stream.write_all(&request_json)?;
    stream.flush()?;

    // 读取响应
    let mut buffer = Vec::new();
    stream.read_to_end(&mut buffer)?;

    let response: DecryptResponse = serde_json::from_slice(&buffer)?;
    println!("解密后的数据: {}", response.decrypted_data);
    println!("公钥: {}", response.public_key);

    Ok(())
} 